pkg/data/types.go

```diff
@@ -140,7 +145,8 @@ type ResourceOffer struct {
 	Spec MachineSpec `json:"spec"`
 	// the module ID's that this resource provider can run
 	// an empty list means ALL modules
-	Modules []string `json:"modules"`
+	Modules []EnabledModules `json:"modules"`
```

Comment:
Can you rename this to `EnabledModules`? `Modules` feels too generic and the intention may get lost in other places where this is used.